lib/_http_server.js

```diff
@@ -281,7 +273,7 @@ function connectionListener(socket) {
   if (self.timeout)
     socket.setTimeout(self.timeout);
   socket.on('timeout', function() {
-    var req = socket.parser && socket.parser.incoming;
+    var req = parser && parser.incoming;
```

Comment:
Yeah, that would be [down there](https://github.com/iojs/io.js/blob/v1.x/lib/_http_server.js#L294). The `on('timeout', ...)` is synchronous so we can avoid the key lookup.